src/Tasks/SystemState.cs

```diff
@@ -210,6 +210,16 @@ internal FrameworkName FrameworkNameAttribute
                 get { return frameworkName; }
                 set { frameworkName = value; }
             }
+
+            /// <summary>
+            /// The last-modified value to use for immutable framework files which we don't do I/O on.
+            /// </summary>
+            internal static DateTime ImmutableFileLastModifiedMarker => DateTime.MaxValue;
+
+            /// <summary>
+            /// It is wasteful to persist entries for immutable framework files.
+            /// </summary>
+            internal bool IsWorthPersisting => lastModified != ImmutableFileLastModifiedMarker;
```

Comment:
If the file doesn't exists, would it be min or max datetime?
Likewise, if the file isn't an assembly?

Reply:
This is a positive cache of assemblies, I don't think we create an entry for a non-existent file or a file that is not an assembly. In general we use `DateTime.MinValue` to represent "missing", though.